nxcomp/Log.h

```diff
@@ -198,18 +286,68 @@ NXLog& operator<< (NXLog& out, const NXLogStamp& value);
 
 
 template <typename T>
+bool has_newline(T value)
+{
+	return false;
+}
+
+template <char*>
+static bool has_newline(char *value)
+{
+	return strstr(value, "\n") != NULL;
```

Comment:
I.e., using the other implementation won't have a positive impact on performance, but at least remove one more function to maintain, because it's a wrapper for a different function.